flask/app/analyses.py

```diff
@@ -178,6 +178,38 @@ def create_analysis():
     except:  # LOGIN DISABLED
         requester_id = updated_by_id = 1
 
+    if app.config.get("LOGIN_DISABLED") or current_user.is_admin:
+        user_id = request.args.get("user")
+    else:
+        user_id = current_user.user_id
+
+    query = (
+        db.session.query(models.Analysis)
+        .options(contains_eager(models.Analysis.datasets))
+        .join(
+            models.datasets_analyses_table,
+            models.Analysis.analysis_id
+            == models.datasets_analyses_table.columns.analysis_id,
+        )
+        .join(models.Dataset)
+        .join(
+            models.groups_datasets_table,
+            models.Dataset.dataset_id
+            == models.groups_datasets_table.columns.dataset_id,
+        )
+        .join(
+            models.users_groups_table,
+            models.groups_datasets_table.columns.group_id
+            == models.users_groups_table.columns.group_id,
+        )
+        .filter(models.users_groups_table.columns.user_id == user_id)
+    )
+
+    permitted_ids = set([x for x, in query.values(models.Dataset.dataset_id)])
+
+    if not all([i in permitted_ids for i in dts_pks]):
+        return "Not all datasets are accessible to the current user", 400
```

Comment:
The reasoning is okay here but it's not very efficient as it takes `O(datasets * requested datasets)` time. We've taken this approach for some `/api/users` and `/api/groups` to have the database server do the hard work instead:
https://github.com/ccmbioinfo/ST2020/blob/4a0595de65e9e043f569cb2a13c8f51d9ac42df9/flask/app/groups.py#L105-L109